### DISCUSSION TITLE
[Question][Hw3] Example1 Expected Coverage

### DISCUSSION THREAD

**Discussion by pacokwon** [prosyslab-classroom/cs348-information-security#521] [Q&A]
I have a question about the `example1.cov.expected` file.
```c
int main() {
  int x = 10;
  int y = 10 + x;
  int z = x - 10;
  int w = y / z;
  return 0;
}
```

In the above example file, it seems to me that there are 4 instructions -
* `add` at line 3
* `sub` at line 4
* `sdiv` at line 5
* `ret` at line 6

And since the above code terminates at line 5, my guess is that the coverage file should have 3 lines - lines 3, 4 **and 5 as well**.
But the `example1.cov.expected` file only has two lines, lines 3 and 4.

Did I misunderstand the question? Any clarifications would be greatly appreciated!

Thank you.

Edit) Maybe there's an implicit requirement that the `__dbz_sanitizer__` function should be called before `__coverage__`? So that it should fail before writing to the file?

**Comment by Junseon00** [comment]
Yes, but as you can see `runtime.c`,  `__dbz_sanitizer__` exit when it has zero on divisor. So it's upto your implementation ,that make the output like `example1.cov.expected`.

**Comment by pacokwon** [comment]
Thanks a lot! Closing
